### PR TITLE
feat(FN-923): update infrastructure with main - vars

### PR DESCRIPTION
**Pull request by OxonianCambion**
### Introduction

We need to be able to manage the infrastructure in declarative IaC - we've chosen Bicep.
The first phase is to be able to:

- produce a complete new deployment environment (feature) mirroring the current ones.
- be in a position to take over the extant environments.
- Note that there have been some manual changes to the environments. This work also functions as an audit of the current infrastructure.

Possible enhancements / updates will be noted but not implemented in the first phase.

This card covers
* updating variable changes from updates to main
### Resolution
* environment variables from secrets updated
* environment variables from vars updated

### CODE REVIEW

File: infrastructure/resource_group_level/modules/webapps/portal-api.bicep [hunk 1]
```diff
@@ -40,19 +40,10 @@ param secureConnectionStrings object = {
   CORS_ORIGIN: 'test-value'
   AZURE_PORTAL_EXPORT_FOLDER: 'test-value'
   AZURE_PORTAL_FILESHARE_NAME: 'test-value'
-  // TODO:FN-749 Confirm *_WORKFLOW_* variables are not needed and can be removed.
```
Comment (OxonianCambion) [inline infrastructure/resource_group_level/modules/webapps/portal-api.bicep:43]: As these were removed in the merge from main, I've removed them too. I'll still double check with FN-749.

File: infrastructure/resource_group_level/modules/webapps/trade-finance-manager-api.bicep [hunk 2]
```diff
@@ -50,6 +48,10 @@ param secureConnectionStrings object = {
 @secure()
 param additionalSecureConnectionStrings object = {
   GOV_NOTIFY_EMAIL_RECIPIENT: 'test-value'
+  // TODO:FN-429 Note that this has a value like https://tfs-dev-tfm-fd.azurefd.net 
+  // while in the CLI it is injected as a secret, we can probably calculate it from the Front Door component.
+  // Note that it has been replaced by TFM_UI_URL in the CLI scripts
+  TFM_URI: 'test-value'
```
Comment (OxonianCambion) [inline infrastructure/resource_group_level/modules/webapps/trade-finance-manager-api.bicep:54]: I suspect this value isn't needed and can go.

File: infrastructure/resource_group_level/modules/webapps/trade-finance-manager-api.bicep [hunk 3]
```diff
@@ -39,9 +39,7 @@ param secureConnectionStrings object = {
   CORS_ORIGIN: 'test-value'
   UKEF_TFM_API_SYSTEM_KEY: 'test-value'
   UKEF_TFM_API_REPORTS_KEY: 'test-value'
-  // TODO:FN-429 Note that this has a value like https://tfs-dev-tfm-fd.azurefd.net 
-  // while in the CLI it is injected as a secret, we can probably calculate it from the Front Door component.
-  TFM_URI: 'test-value'
+  TFM_UI_URL: 'test-value'
```
Comment (OxonianCambion) [inline infrastructure/resource_group_level/modules/webapps/trade-finance-manager-api.bicep:42]: I observed that `TFM_URI` is probably able to be derived. Can this be too, or does it need to come from a secret?

File: infrastructure/resource_group_level/modules/function-acbs.bicep [hunk 4]
```diff
@@ -43,8 +43,12 @@ resource storageAccount 'Microsoft.Storage/storageAccounts@2022-09-01' existing
 
 var storageAccountKey = storageAccount.listKeys().keys[0].value
 
-// These values are hardcoded in the CLI scripts
+// These values are hardcoded in the CLI scripts, derived in the script or set from normal env variables or vars
 var settings = {
+  // from vars.
+  RATE_LIMIT_THRESHOLD: 'test-value'
```
Comment (OxonianCambion) [inline infrastructure/resource_group_level/modules/function-acbs.bicep:49]: I'll update the values that don't come from secrets but should be a parameter shortly, but not now.

File: infrastructure/resource_group_level/modules/webapps/trade-finance-manager-api.bicep (hunk 2 above)
Comment (IgnatG) [reply infrastructure/resource_group_level/modules/webapps/trade-finance-manager-api.bicep:54]: yep, we can remove it in this PR

File: infrastructure/resource_group_level/modules/webapps/trade-finance-manager-api.bicep (hunk 3 above)
Comment (IgnatG) [reply infrastructure/resource_group_level/modules/webapps/trade-finance-manager-api.bicep:42]: can be derived as well

File: infrastructure/resource_group_level/modules/webapps/trade-finance-manager-api.bicep (hunk 2 above)
Comment (OxonianCambion) [reply infrastructure/resource_group_level/modules/webapps/trade-finance-manager-api.bicep:54]: Ok - I will remove it. I think it was changed to `TFM_UI_URL` not the API one, right?

Comment (IgnatG) [reply infrastructure/resource_group_level/modules/webapps/trade-finance-manager-api.bicep:54]: ah yes. you are on TFM API bicep file already. so `TFM_UI_URL` it is

Comment (OxonianCambion) [reply infrastructure/resource_group_level/modules/webapps/trade-finance-manager-api.bicep:54]: ok! Note that the change has not been made in the configuration - I can still see `TFM_URI` in the connection strings area and not `TFM_UI_URL` (or even `TFM_API_URL`).

File: infrastructure/resource_group_level/modules/webapps/trade-finance-manager-api.bicep (hunk 3 above)
Comment (OxonianCambion) [reply infrastructure/resource_group_level/modules/webapps/trade-finance-manager-api.bicep:42]: Oh wait; I see why it was set via a secret - there's a circular dependency.
`TfmApi` -> `TfmUi` -> `ApplicationGratewayTfm` -> `TfmFrontdoor` -> `TfmApi`.

I'll make an explicit card to think about that.

**Review by IgnatG — APPROVED** [review]

**Pull request merged**